server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -582,21 +595,47 @@ private boolean isExpired(Subscription subscription) {
     }
 
     @Transactional
-    private void deleteExcessEntitlements(Pool existingPool) {
+    private void deleteExcessEntitlements(List<Pool> existingPools) {
         boolean lifo = !config
             .getBoolean(ConfigProperties.REVOKE_ENTITLEMENT_IN_FIFO_ORDER);
+        if (existingPools != null && !existingPools.isEmpty()) {
```

Comment:
Changed.
However, I think generally speaking folks are split on this, there are pros and cons.
simply said, reducing nests vs. risking logic errors introduced by having more than one exit points.

I dont feel strongly about either, so changed it.